src/WebJobs.Extensions.DurableTask/Listener/FunctionExecutionHelper.cs

```diff
@@ -26,27 +27,40 @@ internal static class FunctionExecutionHelper
             try
             {
                 context.ExecutorCalledBack = false;
+                var timeoutSource = new System.Threading.Tasks.TaskCompletionSource<Exception>();
+                shim.TimeoutTask = timeoutSource.Task;
```

Comment:
Any reason not to always assign `TimeoutTask` when the shim object is created? I'm wondering if we need to be careful about potential null-ref exceptions for consumers of this property since it's not immediately obvious when this property gets assigned (this is a situation where nullable reference type annotations in C# can be really helpful).

Reply:
Makes sense, I will change that.